src/config.ts

```diff
@@ -0,0 +1,94 @@
+import yargs from "yargs";
+import { readFileSync } from 'fs';
+import { Address } from "@polkadot/types/interfaces/runtime";
+
+import { EmailConfig } from './reporters/email';
+import { MatrixConfig } from './reporters/matrix';
+
+import { Reporter, ConsoleReporter, EmailReporter, MatrixReporter } from "./reporters";
+
+export interface ExtendedAccount {
+    address: Address,
+    label: string,
+}
+
+export interface AppConfig {
+    endpoints: string[],
+    accounts: ExtendedAccount[],
+    eventFilter: "all" | string[],
+    matrix?: MatrixConfig,
```

Comment:
```
reporters: {
  email: ..,
  matrix: ..,
  console: ..,
}
```

is better since all of them are nested in `reporters`. For example. how can your app now log how many reporters it has? it can't.

Reply:
Agreed, done